scrapy/exporters.py

```diff
@@ -363,3 +363,34 @@ def export_item(self, item):
         if self.binary:
             result = dict(self._serialize_item(result))
         return result
+
+
+class ItemExporterFactory:
```

Comment:
I don't see how is this factory useful to be honest.

Reply:
This PR has academic purposes (university project) but also allows the user to create a different exporter at runtime thanks to the type parameter

Reply:
Please don't submit PRs that only have a purpose of demonstrating to your teacher that you wrote some code.